src/extendedstorage.cpp

```diff
@@ -671,10 +672,18 @@ void ExtendedStorage::setAlarms( const Incidence::Ptr &incidence )
       }
       e.setAttribute( "type", "todo" );
     } else if ( incidence->dtStart().isValid() ) {
-      e.setAttribute( "time", incidence->dtStart().toString() );
-      e.setAttribute( "startDate", incidence->dtStart().toString() );
-      if (incidence->endDateForStart( incidence->dtStart()).isValid() ) {
-        e.setAttribute( "endDate", incidence->endDateForStart( incidence->dtStart()).toString() );
+      KDateTime eventStart;
+
+      if (incidence->recurs()) {
+        // assuming alarms not later than event start
+        eventStart = incidence->recurrence()->getNextDateTime( alarmTime.addSecs(-60) );
+      } else {
+        eventStart = incidence->dtStart();
+      }
+      e.setAttribute( "time", eventStart.toString() );
+      e.setAttribute( "startDate", eventStart.toString() );
+      if (incidence->endDateForStart( eventStart).isValid() ) {
+        e.setAttribute( "endDate", incidence->endDateForStart( eventStart ).toString() );
```

Comment:
This was actually old code, but while at it, I'll change that and above recurs() to follow ( x ) as being more common in mkcal.